cumulus/primitives/core/src/lib.rs

```diff
@@ -78,9 +78,18 @@ impl From<MessageSendError> for &'static str {
 	}
 }
 
-/// The origin of an inbound message.
+/// The aggregate origin of an inbound message.
 #[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
 pub enum AggregateMessageOrigin {
+	/// The message came from the DMP or HRMP subsystem
```

Comment:
I dont think we can just change this enum, since its use by the MQ pallet as queue ID.  
That would mean that we need a lazy migration across all parachain which is a huge pain.  
It would be better to just add a variant to it like `Abridged` or `Other` and keeping the others compatible.

Reply:
Unless this change goes in directly after https://github.com/paritytech/polkadot-sdk/pull/1246 in the same release, but its difficult.

Reply:
Also this doesn't make sense, `XCM` is not an origin or transport protocol. I can't tell if you are indicating the _format_ of the message or its origin with this type? If the latter, why not just add a new variant to the `AggregateMessageOrigin` enum, like `ExternalConsensus(Junction)`, where the `Junction` is expected to be of the variant `GlobalConsensus(NetworkId)`?

Reply:
> I dont think we can just change this enum, since its use by the MQ pallet as queue ID.

Good point, didn't think of that. I'll will just add another variant to the enum then.

Here's a new design, inline with the above suggested changes.

Since the `NetworkId` and `Junction` types are not stable across XCM versions (and because they are used at higher protocol layers), I have not used them here.

```rust
/// The origin of an inbound message.
#[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
pub enum AggregateMessageOrigin {
	/// The message came from the para-chain itself.
	Here,
	/// The message came from the relay-chain.
	///
	/// This is used by the DMP queue.
	Parent,
	/// The message came from a sibling para-chain.
	///
	/// This is used by the HRMP queue.
	Sibling(ParaId),
	/// External consensus
	ExternalConsensus {
		origin: ExternalConsensusOrigin,
		dest: ExternalConsensusDestination
	}
}

/// External consensus destinations
#[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
pub enum ExternalConsensusDestination {
	// This network is reachable via Snowbridge on BridgeHub
	Ethereum { chain_id: u64 }
}

/// The origin of a message destined for an external consensus system
#[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
pub enum ExternalConsensusOrigin {
	Here,
	Sibling(ParaId)
}
```